pkg/redshift/api/api_test.go

```diff
@@ -216,3 +216,51 @@ func Test_GetSecret(t *testing.T) {
 		t.Errorf("unexpected result: %v", cmp.Diff(expectedSecret, secret))
 	}
 }
+
+func Test_GetCluster(t *testing.T) {
+	c := &API{ManagementClient: &redshiftclientmock.MockRedshiftClient{Clusters: []string{"foo"}}}
+	cluster, err := c.Cluster(sqlds.Options{"clusterIdentifier": "foo"})
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+	expectedCluster := &models.RedshiftCluster{
+		Endpoint: models.RedshiftEndpoint{ 
+			Address: "foo", 
+			Port: 123,
+		}, 
+		Database: "foo",
+	}
+	if !cmp.Equal(expectedCluster, cluster) {
+		t.Errorf("unexpected result: %v", cmp.Diff(expectedCluster, cluster))
+	}
+}
+
+func Test_GetCluster_Errors(t *testing.T) {
+	fooC := &API{ManagementClient: &redshiftclientmock.MockRedshiftClient{Clusters: []string{"foo"}}}
+
+	tests := []struct {
+		c		  *API
+		desc	  string
+		clusterId string
+		errMsg    string
+	}{
+		{
+			c: fooC,
+			desc: "Error cluster ID not found",
+			clusterId: "xyz",
+			errMsg: "ClusterId xyz not found",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			c := tt.c
+			cluster, err := c.Cluster(sqlds.Options{"clusterIdentifier": tt.clusterId})
+			if (err.Error() != tt.errMsg) {
```

Comment:
`err` may be `nil` too, so you need to check that to avoid a panic in the test.

Reply:
also, you can simplify these checks with the [`assert` package](https://pkg.go.dev/github.com/stretchr/testify/assert#EqualError):

```
			assert.Nil(t, cluster)
			assert.EqualError(t, err, tt.errMsg)
```

will do what you want